oauth2client/service_account.py

```diff
@@ -61,31 +61,50 @@ def __init__(self, service_account_id, service_account_email, private_key_id,
   def _generate_assertion(self):
```

Comment:
I was hoping someone could elucidate for me where that happens in the codebase as part of guidance for writing the tests...